src/library_webgl2.js

```diff
@@ -1110,10 +1110,13 @@ var webgl2Funcs = [[0, 'endTransformFeedback pauseTransformFeedback resumeTransf
 
 #if USE_WEBGL2
 
-createGLPassthroughFunctions(LibraryWebGL2, webgl2Funcs);
+// If user passes -s USE_WEBGL2=1 -s STRICT=1 but not -lGL (to link in WebGL 1), then ignore linking in WebGL 2 either.
+if (typeof createGLPassthroughFunctions !== 'undefined') {
```

Comment:
this check seems a little odd here. how about in emcc.py, if strict and no -lGL then disabling `USE_WEBGL*`?

Reply:
Hmm, I can do that if preferred, though in my mind this kind of check is more preferable since it is more self-contained/cohesive?

Reply:
What about some kind of "error: USE_WEBGL2 doesn't work without -lGL"?  Or have USE_WEBGL2 imply -lGL?  Having it silently not work seems wrong.

Reply:
I agree it's more self-contained your way. I suspect that there might be other benefits to doing it cleanly in emcc.py, though (that is, other side effects of `USE_WEBGL` would be removed if we disable it at the source). But if you think that's unlikely then lgtm as is.

Reply:
Ok, updated with an error message in library_webgl2.js, that should address both comments.